src/cpu_shader/pathtag_reduce.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 The Vello authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use vello_encoding::{ConfigUniform, Monoid, PathMonoid};
+
+use crate::cpu_dispatch::CpuBinding;
+
+const WG_SIZE: usize = 256;
+
+fn pathtag_reduce_main(
+    n_wg: u32,
+    config: &ConfigUniform,
+    scene: &[u32],
+    reduced: &mut [PathMonoid],
+) {
+    let pathtag_base = config.layout.path_tag_base;
+    for i in 0..n_wg {
+        let mut m = PathMonoid::default();
+        for j in 0..WG_SIZE {
+            let tag = scene[(pathtag_base + i * WG_SIZE as u32) as usize + j];
```

Comment:
`(pathtag_base + i * WG_SIZE as u32) as usize` is a constant per loop iteration, right?

Obviously, this should get hoisted anyway, but it might be clearer to split it out